pkg/providers/gcp/membrane.go

```diff
@@ -29,14 +42,57 @@ import (
 	secret_manager_secret_service "github.com/nitrictech/nitric/pkg/plugins/secret/secret_manager"
 	storage_service "github.com/nitrictech/nitric/pkg/plugins/storage/storage"
 	"github.com/nitrictech/nitric/pkg/providers/gcp/core"
+	"github.com/nitrictech/nitric/pkg/span"
+	"github.com/nitrictech/nitric/pkg/utils"
 )
 
+func newTraceProvider(ctx context.Context) (*sdktrace.TracerProvider, error) {
+	span.FunctionName = os.Getenv("K_SERVICE")
+	span.UseFuncNameAsSpanName = false
+
+	exp, err := otlptracegrpc.New(ctx, otlptracegrpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := resource.New(ctx,
+		resource.WithDetectors(gcp.NewDetector()),
+		resource.WithAttributes(
+			semconv.CloudProviderGCP,
+			semconv.CloudPlatformGCPCloudRun,
+			attribute.Key("component").String("Nitric membrane"),
+			semconv.ServiceNameKey.String(span.FunctionName),
+			semconv.ServiceNamespaceKey.String(utils.GetEnv("NITRIC_STACK", "")),
+		),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	otel.SetTextMapPropagator(
+		propagation.NewCompositeTextMapPropagator(
+			propagator.CloudTraceFormatPropagator{},
+			propagation.TraceContext{},
+			propagation.Baggage{},
+		))
+
+	rate, err := utils.PercentFromIntString(utils.GetEnv("NITRIC_TRACE_SAMPLE_PERCENT", "10"))
+	if err != nil {
+		return nil, errors.WithMessagef(err, "NITRIC_TRACE_SAMPLE_PERCENT should be an int not %s", rate)
+	}
+
+	return sdktrace.NewTracerProvider(
+		sdktrace.WithSampler(sdktrace.ParentBased(sdktrace.TraceIDRatioBased(rate))),
+		sdktrace.WithResource(res),
+		sdktrace.WithBatcher(exp),
+	), nil
+}
```

Comment:
would be good to do this consistently, probably in a separate file like you have done for aws.